server/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -71,7 +73,37 @@ public TransportValidateQueryAction(Settings settings, ThreadPool threadPool, Cl
     @Override
     protected void doExecute(Task task, ValidateQueryRequest request, ActionListener<ValidateQueryResponse> listener) {
         request.nowInMillis = System.currentTimeMillis();
-        super.doExecute(task, request, listener);
+        LongSupplier timeProvider = () -> request.nowInMillis;
+        ActionListener<org.elasticsearch.index.query.QueryBuilder> rewriteListener = ActionListener.wrap(rewrittenQuery -> {
+            request.query(rewrittenQuery);
+            super.doExecute(task, request, listener);
+        }, ex -> {
```

Comment:
I added this section (with advice from @imotov) to address a deficiency in the code uncovered by one of our integration tests, `org.elasticsearch.validate.SimpleValidateQueryIT.testExplainValidateQueryTwoNodes`, which tests an error case. Because originally I was propagating the listener's `onError` method, an XContent exception was being thrown by the server, rather than wrapped in a response that wraps the error according to our conventions (yay, integration tests!).

My "fix" raises a few questions about what the right response looks like, given this context.